src/zx/FunctionalityConstruction.cpp

```diff
@@ -0,0 +1,425 @@
+#include "zx/FunctionalityConstruction.hpp"
+
+#include "Definitions.hpp"
+#include "ZXDiagram.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace zx {
+
+    bool FunctionalityConstruction::checkSwap(op_it it, op_it end, Qubit ctrl,
+                                              Qubit target, const qc::Permutation& p) {
```

Comment:
Not that it matters that much, but I believe both iterators and the Qubits can be made `const` here.
That might hold for more of the parameters down below.